examples/print.js

```diff
@@ -0,0 +1,245 @@
+ * @private
+ */
+app.PRINT_PAPER_SIZE_ = [555, 675];
+
+
+
+/**
+ * @constructor
+ * @param {angular.$timeout} $timeout Angular timeout service.
+ * @param {ngeo.CreatePrint} ngeoCreatePrint The ngeo Create Print function.
+ * @param {ngeo.PrintUtils} ngeoPrintUtils The ngeo PrintUtils service.
+ * @ngInject
+ * @export
+ */
+app.MainController = function($timeout, ngeoCreatePrint, ngeoPrintUtils) {
+
+  var projection = ol.proj.get('EPSG:21781');
+  projection.setExtent([485869.5728, 76443.1884, 837076.5648, 299941.7864]);
+
+  /**
+   * @type {ol.Map}
+   */
+  this['map'] = new ol.Map({
+    layers: [
+      new ol.layer.Image({
+        source: new ol.source.ImageWMS({
+          url: app.WMS_URL_,
+          params: {
+            'LAYERS': 'osm'
+          },
+          serverType: /** @type {ol.source.wms.ServerType} */ ('mapserver')
+        })
+      })
+    ],
+    view: new ol.View({
+      projection: projection,
+      resolutions: [50, 20, 10, 5, 2.5, 2, 1],
+      center: [537635, 152640],
+      zoom: 3
+    })
+  });
+
+  /**
+   * Text to display a "loading" message while waiting for the report.
+   * @type {string}
+   */
+  this['printState'] = '';
+
+  /**
+   * @type {angular.$timeout}
+   * @private
+   */
+  this.$timeout_ = $timeout;
+
+  /**
+   * @type {ngeo.Print}
+   * @private
+   */
+  this.print_ = ngeoCreatePrint(app.PRINT_URL_);
+
+  /**
+   * @type {ngeo.PrintUtils}
+   * @private
+   */
+  this.printUtils_ = ngeoPrintUtils;
+
+  /**
+   * @type {function(ol.render.Event)}
+   */
+  var postcomposeListener = ngeoPrintUtils.createPrintMaskPostcompose(
+      /**
+       * @return {ol.Size} Size in dots of the map to print.
+       */
+      function() {
+        return app.PRINT_PAPER_SIZE_;
+      },
+      /**
+       * @param {olx.FrameState} frameState Frame state.
+       * @return {number} Scale of the map to print.
+       */
+      function(frameState) {
+        var mapSize = frameState.size;
+        var mapResolution = frameState.viewState.resolution;
+        return ngeoPrintUtils.getOptimalScale(mapSize, mapResolution,
+            app.PRINT_PAPER_SIZE_, app.PRINT_SCALES_);
+      });
+
+  /**
+   * Draw the print window in a map postcompose listener.
+   */
+  this['map'].on('postcompose', postcomposeListener);
+};
+
+
+/**
+ * @export
+ */
+app.MainController.prototype.print = function() {
+  var map = this['map'];
+
+  var mapSize = map.getSize();
+  var viewResolution = map.getView().getResolution();
+
+  var scale = this.printUtils_.getOptimalScale(mapSize, viewResolution,
+      app.PRINT_PAPER_SIZE_, app.PRINT_SCALES_);
+
+  var dpi = app.PRINT_DPI_;
+  var layout = app.PRINT_LAYOUT_;
+
+  this['printState'] = 'Printing…';
+
+  var spec = this.print_.createSpec(map, scale, dpi, layout, {
+    'datasource': [],
+    'debug': 0,
+    'comments': 'My comments',
+    'title': 'My print'
+  });
+
+  this.print_.createReport(spec).then(
+      angular.bind(this, this.handleCreateReportSuccess_),
```

Comment:
This is explained in the [developer guide](https://github.com/camptocamp/ngeo/blob/master/docs/developer-guide.md#authoring-examples):

« As described above the gh-pages make target can be used to publish the examples to github.io. The examples published on github.io are not compiled, they rely on the standalone ngeo.js build.

Because of that the examples cannot rely on non-exported symbols or properties. For example they cannot use goog objects and they cannot use ol objects that are not part of the OpenLayers API (that is marked with @api in the OpenLayers source code). »